packages/app-store/routing-forms/pages/route-builder/[...appPages].tsx

```diff
@@ -280,23 +282,29 @@ const Route = ({
                         }
                         if (option.value !== "custom") {
                           setRoute(route.id, { action: { ...route.action, value: option.value } });
+                          setCustomEventTypeSlug("");
                         } else {
                           setRoute(route.id, { action: { ...route.action, value: "custom" } });
                           setCustomEventTypeSlug("");
                         }
                       }}
                       value={
                         eventOptions.length !== 0 && route.action.value !== ""
-                          ? eventOptions.find((eventOption) => eventOption.value === route.action.value) || {
+                          ? eventOptions.find(
+                              (eventOption) =>
+                                eventOption.value === route.action.value && !customEventTypeSlug.length
+                            ) || {
                               label: t("custom"),
                               value: "custom",
                             }
                           : undefined
                       }
                     />
                     {eventOptions.length !== 0 &&
-                      route.action.value !== "" &&
-                      !eventOptions.find((eventOption) => eventOption.value === route.action.value) && (
+                    route.action.value !== "" &&
+                    (!eventOptions.find((eventOption) => eventOption.value === route.action.value) ||
+                      customEventTypeSlug.length) ? (
```

Comment:
related to: https://github.com/calcom/cal.com/pull/13624/files#diff-243a8c40cc62828c5dc7d4683aee4d4541c773c7556937af97f204838d5a82ddR295

Textfield wasn't shown no more when `customEventTypeSlug` was the same as an existing event type slug